Monika After Story/game/definitions.rpy

```diff
@@ -5,6 +5,102 @@ define config.developer = False #This is the flag for Developer tools
 init 1 python:
     persistent.steam = "steamapps" in config.basedir.lower()
 
+### Overrides of core renpy things
+python early in mas_overrides:
+    from store import *
+
+    def dummy(*args, **kwargs):
+        """
+        Dummy function that does nothing
+        """
+        return
+
+    # clear this so no more traceback. We expect node loops anyway
+    renpy.execution.check_infinite_loop = dummy
+
+    def get_field_override(self, field_name, args, kwargs):
+        """
+        Originally this method returns objects by references
+        This override allows us to eval functions when appropriate, e.g. "Text [myfunc(arg1)]."
+        and use negative indexes for iterables, e.g. "Text [_list[-2]]."
+
+        IN:
+            field_name - the reference to the object
+            args - not sure, but renpy doesn't use it (passes in an empty tuple)
+            kwargs - the store modules where renpy will look for the object
+
+        OUT:
+            tuple of the object and its key
+        """
+        # if it's a function, we eval it
+        if "(" in field_name:
+            # id to split the string
+            split_index = field_name.index("(")
+            # get the func's name
+            func_name = field_name[:split_index]
+            # get the arg and kwargs
+            args = field_name[split_index:]
+
+            # it still may include store modules, get the 1st if there's any
+            if "." in func_name:
+                func_store_name, func_name = func_name.split(".", 1)
+                first = func_store_name
+                func_store_name += "."
+
+            else:
+                func_store_name = ""
+                first = func_name
+
+            # now we find the store's name to use in eval
+            # iterate through all the scopes you passed in to find the one
+            # where the function/module was defined
+            if isinstance(kwargs, renpy.substitutions.MultipleDict):
+                for scope in kwargs.dicts:
+                    if first in scope:
+                        stores_names_list = [
+                            store_obj_name
+                            for store_obj_name, store_obj in sys.modules.iteritems()
+                            if store_obj and store_obj.__dict__ is scope
+                        ]
+                        if stores_names_list:
+                            scope_store_name = "renpy.{0}.".format(stores_names_list[0])
+                            break
```

Comment:
don't use `break`, please. (or a `for-else` loop, which I have actually never seen before, but i would argue is less readable as its easy to to mistake the `else` as part of a preceding `if`). Make this a separate function so it could be used elsewhere and use `return`.